### ISSUE TITLE
detci diag_method=rsp not working

### ISSUE THREAD

**Issue by susilehtola** [psi4/psi4#558]
Given the following input
```
molecule {
symmetry c1
He 0.0 0.0 0.0
He 0.0 0.0 1000.0
}

set basis 6-31G
set GUESS_VECTOR UNIT
set NUM_DETS_PRINT 4000000000000
set e_convergence 1e-10
set r_convergence 1e-10
#set diag_method rsp
energy('fci')
```

I get the same energy with SEM and RSP, but the wave functions are totally different: SEM yields
```
   The 36 most important determinants:

    *   1   -0.995668  (    0,    0)  1AX  2AX  
    *   2    0.065607  (    3,    3)  1AX  4AX  
    *   3    0.065607  (    2,    2)  2AX  3AX  
    *   4   -0.004323  (    5,    5)  3AX  4AX  
    *   5    0.002076  (    0,    3)  1AX  2AA  4AB  
    *   6    0.002076  (    3,    0)  1AX  2AB  4AA  
    *   7   -0.002076  (    0,    2)  1AA  2AX  3AB  
    *   8   -0.002076  (    2,    0)  1AB  2AX  3AA  
    *   9    0.000137  (    2,    5)  2AA  3AX  4AB  
    *  10    0.000137  (    5,    2)  2AB  3AX  4AA  
    *  11   -0.000137  (    3,    5)  1AA  3AB  4AX  
    *  12   -0.000137  (    5,    3)  1AB  3AA  4AX  
    *  13    0.000004  (    2,    3)  1AB  2AA  3AA  4AB  
    *  14    0.000004  (    3,    2)  1AA  2AB  3AB  4AA  
    *  15   -0.000004  (    0,    5)  1AA  2AA  3AB  4AB  
    *  16   -0.000004  (    5,    0)  1AB  2AB  3AA  4AA  
```

while RSP gives
```
   The 36 most important determinants:

    *   1    0.995668  (    0,    0)  1AX  2AX  
    *   2    0.089327  (    2,    3)  1AB  2AA  3AA  4AB  
    *   3   -0.023675  (    2,    2)  2AX  3AX  
    *   4    0.009265  (    0,    4)  1AA  2AX  4AB  
    *   5    0.004289  (    5,    5)  3AX  4AX  
    *   6   -0.000549  (    4,    4)  2AX  4AX  
    *   7    0.000262  (    4,    3)  1AB  2AA  4AX  
    *   8    0.000159  (    0,    3)  1AX  2AA  4AB  
    *   9   -0.000043  (    2,    4)  2AX  3AA  4AB  
```

The RSP solution is obviously wrong, since the quadruple excitation should be possible to write in terms of independent singles/doubles.

**Comment by dgasmith** [comment]
This is a pretty odd error. Examining the Hamiltonian built by RSP is fine; however, if you diagonalize it with Mac Accelerate I get your `RSP` result while using MKL DSYEV I get the `SEM` result. The eigenvalues are fine, just looks like something is going wrong with the eigenvectors.

**Comment by dgasmith** [comment]
Looks like I was getting my tests confused when running your issues. The eigenvectors are indeed wrong. Fixed in #566.
